AGP version compatibility for Cordova-Android 12

# Bug report

**CHECKLIST**
- [x] I have read the [issue reporting guidelines](https://github.com/dpa99c/cordova-plugin-firebasex#reporting-issues)

- [x] I confirm this is a suspected bug or issue that will affect other users

- [x] I have reproduced the issue using the [example project](https://github.com/dpa99c/cordova-plugin-firebasex-test) or provided the necessary information to reproduce the issue.

- [x] I have read [the documentation](https://github.com/dpa99c/cordova-plugin-firebasex) thoroughly and it does not help solve my issue.

- [x] I have checked that no similar issues (open or closed) already exist.



**Current behavior:**
An exception is happening during build phase using new version Cordova Android 12.

**Expected behavior:**
Build to work

**Steps to reproduce:**
cordova platform rm android
cordova platform add android@12

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->

**Environment information**
<!-- Please supply full details of your development environment including: -->
- Cordova CLI version 
	- 11.1.0
- Cordova platform version
	- Installed platforms:
  android 12.0.0
- Plugins & versions installed in project (including this plugin)
cordova-plugin-actionsheet 2.3.3 "ActionSheet"
cordova-plugin-androidx-adapter 1.1.3 "cordova-plugin-androidx-adapter"
cordova-plugin-app-version 0.1.14 "AppVersion"
cordova-plugin-device 2.1.0 "Device"
cordova-plugin-dialogs 2.0.2 "Notification"
cordova-plugin-firebasex 16.0.0 "Google Firebase Plugin"
cordova-plugin-inappbrowser 5.0.0 "InAppBrowser"
cordova-plugin-ionic-keyboard 2.2.0 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 4.1.3 "cordova-plugin-ionic-webview"
cordova-plugin-market 1.2.0 "Market"
cordova-plugin-network-information 3.0.0 "Network Information"
cordova-plugin-statusbar 2.4.3 "StatusBar"
cordova-sqlite-storage 6.1.0 "Cordova SQLite storage plugin - cordova-sqlite-storage plugin version"
phonegap-plugin-barcodescanner 8.1.1-dev "BarcodeScanner"
uk.co.workingedge.phonegap.plugin.launchnavigator 5.0.6 "Launch Navigator"
- Dev machine OS and version, e.g.
    - OSX
        - `sw_vers`
    - Windows 10
        - `winver`
        
_Runtime issue_
- Device details
    - _e.g. iPhone X, Samsung Galaxy S8, iPhone X Simulator, Pixel XL Emulator_
- OS details
    - _e.g. iOS 12.2, Android 9.0_	
	
_Android build issue:_	
- Node JS version
    - `node -v`
- Gradle version
	- 7.6
- Target Android SDK version
	- 33
- Android SDK details
Installed packages:
  Path                 | Version | Description                    | Location
  -------              | ------- | -------                        | -------
  build-tools;30.0.3   | 30.0.3  | Android SDK Build-Tools 30.0.3 | build-tools/30.0.3
  build-tools;32.0.0   | 32.0.0  | Android SDK Build-Tools 32     | build-tools/32.0.0
  build-tools;33.0.2   | 33.0.2  | Android SDK Build-Tools 33.0.2 | build-tools/33.0.2
  platform-tools       | 34.0.1  | Android SDK Platform-Tools     | platform-tools
  platforms;android-29 | 5       | Android SDK Platform 29        | platforms/android-29
  platforms;android-30 | 3       | Android SDK Platform 30        | platforms/android-30
  platforms;android-32 | 1       | Android SDK Platform 32        | platforms/android-32
  platforms;android-33 | 2       | Android SDK Platform 33        | platforms/android-33
  tools                | 25.0.0  | Android SDK Tools              | tools
	
_iOS build issue:_
- Node JS version
    - `node -v`
- XCode version


**Related code:**
I will create a PR, but I guess it will be nice to use cordova android variable for AGP version:
classpath "com.android.tools.build:gradle:${cordovaConfig.AGP_VERSION}"
classpath "com.google.gms:google-services:${cordovaConfig.GRADLE_PLUGIN_GOOGLE_SERVICES_VERSION}"

**Console output**

```
> Task :app:mapDebugSourceSetPaths FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mapDebugSourceSetPaths'.
> Error while evaluating property 'extraGeneratedResDir' of task ':app:mapDebugSourceSetPaths'.
   > Failed to calculate the value of task ':app:mapDebugSourceSetPaths' property 'extraGeneratedResDir'.
      > Querying the mapped value of provider(java.util.Set) before task ':app:processDebugGoogleServices' has completed is not supported

```


**Other information:**

On version Cordova Android, it uses :
com.android.tools.build:gradle:7.4.2
com.google.gms:google-services:4.3.15

to be compatible with the change of gradle version I guess

<!--
A POLITE REMINDER

- This is free, open-source software. 
- Although the author makes every effort to maintain it, no guarantees are made as to the quality or reliability, and reported issues will be addressed if and when the author has time. 
- Help/support will not be given by the author, so forums (e.g. Ionic) or Stack Overflow should be used. Any issues requesting help/support will be closed immediately.
- If you have urgent need of a bug fix/feature, the author can be engaged for PAID contract work to do so: please contact dave@workingedge.co.uk
- Rude or abusive comments/issues will not be tolerated, nor will opening multiple issues if those previously closed are deemed unsuitable. Any of the above will result in you being BANNED from ALL of my Github repositories.
-->

--- a/src/android/build.gradle
+++ b/src/android/build.gradle
@@ -5,9 +5,9 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.4'
-        classpath 'com.google.gms:google-services:4.3.10'
-        classpath 'com.google.firebase:firebase-crashlytics-gradle:2.8.1'
+        classpath "com.android.tools.build:gradle:${cordovaConfig.AGP_VERSION}"
+        classpath "com.google.gms:google-services:${cordovaConfig.GRADLE_PLUGIN_GOOGLE_SERVICES_VERSION}"
+        classpath "com.google.firebase:firebase-crashlytics-gradle:2.8.1"
     }
 }
 repositories {

